library/classes/Document.class.php

```diff
@@ -981,6 +990,27 @@ function createDocument(
             $this->couch_docid = $docid;
             $this->couch_revid = $revid;
         } else {
+            /**
+             * https://www.php.net/manual/en/language.fibers.php
+             * Fibers provide a way to pause a running function and return to it later.
+             * In this application of the Fiber class will be used to halt the local file storage.
+             * A dispatch will be sent to see if any module is listening for the file upload.
+             * If a listener is found then the file will be uploaded to the listener and the fiber will be terminated.
+             * If no listener is found then the file will be stored locally.
+             * First use of fiber Sherwin Gaddis 2/13/2024
+             */
+            $fiber = new Fiber(function () use (
```

Comment:
I don't understand what the Fiber is gaining you here. How is this any different than just firing the event on line 1002 and checking if the event handled the upload? Are you trying to avoid a nested if statement here?  That could be accomplished just by pulling the lines in this code into a separate function. There's no real parallelism here using the fiber since you're just immediately suspending it and you're not using it in the dispatched event.  I'm scratching my head what this gains you here.  Perhaps someone else can review this and help me understand what this gains you.  I thought perhaps you were trying to do some parallel processing but I don't see that happening here.

Reply:
It seems like a good idea at the time.

Reply:
I will go back to the drawing board and rethink this.. :-(